Allow json and yaml for file backend

andrewheberle opened the pull request:
This PR adds a check in the file backend and performs a YAML or JSON unmarshal depending on the file extension

abtreece commented:
Hey @andrewheberle 

It appears that the import should be `path/filepath`

```
index 6ee8fa4..24f0620 100644
--- a/backends/file/client.go
+++ b/backends/file/client.go
@@ -2,7 +2,7 @@ package file

 import (
        "encoding/json"
-       "filepath"
+       "path/filepath"
        "fmt"
        "io/ioutil"
        "path"
```

Also, can you add integration tests for .json files... `integration/file/test.sh`

andrewheberle commented:
I've fixed that error and added the tests, which uncovered a problem with the original code, so there are a few more changes in `backends/file/client.go`
